corehq/project_limits/gauge.py

```diff
@@ -86,3 +92,95 @@ def avg(self):
 
 
 case_pillow_lag_gauge = Gauge(feature_key=CASE_SQL, scope_fn=get_all_kafka_partitons_for_topic)
+
+
+class GaugeLimiter:
+
+    def __init__(self, gauge, get_gauge_definition):
+        """
+        :param gauge: ``Gauge`` An instance of Gauge class
+        :param get_gauge_defintions: ``function``
+            A function to get config that GaugeLimiter will use to
+            figure out throttling conditions.
+        """
+        self.gauge = gauge
+        self.gauge_definition_fn = get_gauge_definition
+
+    def wait(self):
+        if not self.allow_usage():
+            time.sleep(self.gauge_definition.throttle_for_seconds)
+
+    def allow_usage(self):
+        if not self.gauge_definition:
+            return False
```

Comment:
Looks like this disallows usage if there is no gauge definition, which does not seem like a reasonable default (assuming absence of gauge definition would be a default state). Am I thinking about that correctly?

Reply:
The code is incorrect. It should have returned `True`. During refactoring I think I just updated the function in the child class and missed the base class. Thanks for pointing it out.

Reply:
[582906f](https://github.com/dimagi/commcare-hq/pull/34800/commits/582906f728aea7fec7b69ea71830bcfa7bf7ac2b)